CI: Migrate to windows-2022

GitHub is starting a migration of `windows-latest` to `windows-2022`:
https://github.blog/changelog/2022-01-11-github-actions-jobs-running-on-windows-latest-are-now-running-on-windows-server-2022/
https://github.com/actions/virtual-environments/issues/4856

Migration starts now, and should finish by March 6. It is not clear if that timetable includes the `-xl` runners.  Based on the fact that the `ubuntu-latest-xl` runners have not yet migrated to 20.04, which was "completed" in 2021-03-18 (and it is now 2022-01-15), the timing for the `-xl` images might be much slower.

UPDATE from GitHub: Our first goal is to GA Windows 2022 this quarter and then we will start the process of getting the XL pools ready.

It may be possible to lock to an older version with `windows-2019-xl` (as it appears those version-specific qualifiers work with ubuntu), but I have not tested that.

I have started testing some builds in 2022 and they are not going well, I'll try to document some issues here.

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"dpaoliello"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

Issue: LNK1318

While building compiletest, it failed to link with this error:

```
Building stage0 tool compiletest (x86_64-pc-windows-msvc)
   Compiling winapi v0.3.9
   Compiling proc-macro2 v1.0.30
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.80
   Compiling cfg-if v1.0.0
   Compiling lazy_static v1.4.0
   Compiling memchr v2.4.1
   Compiling serde_derive v1.0.125
   Compiling ryu v1.0.5
error: linking with `link.exe` failed: exit code: 1201
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.30.30705\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.0.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.1.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.10.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.11.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.12.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.13.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.14.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.15.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.2.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.3.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.4.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.5.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.6.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.7.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.8.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.build_script_build.bca27ebc-cgu.9.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.3w7fvgqa73wuqh0q.rcgu.o" "/LIBPATH:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\deps" "/LIBPATH:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-452d1b1b45f75fb9.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-5ceab4181bd1dfb1.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-ee63620c9a615dfc.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-1dced171ac40f317.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-e344794c4402c589.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-4a873b5445642af4.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-b457ffb0220370f3.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-3dcf84390a40067d.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-41cfe43f2c15c320.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-32a64f0768a3b3dc.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-0a994057c9b22b29.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-7ee144391913f95b.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-a1bb578584eeed3b.rlib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "advapi32.lib" "userenv.lib" "kernel32.lib" "libcmt.lib" "/NXCOMPAT" "/LIBPATH:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\serde_derive-32b276e50e97e8da\\build_script_build-32b276e50e97e8da.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\libstd.natvis"
  = note: LINK : fatal error LNK1201: error writing to program database ''; check for insufficient disk space, invalid path, or insufficient privilege

          

error: could not compile `serde_derive` due to previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `link.exe` failed: exit code: 1201
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.30.30705\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.0.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.1.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.10.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.11.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.12.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.13.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.14.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.15.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.2.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.3.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.4.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.5.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.6.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.7.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.8.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.build_script_build.69b53e7b-cgu.9.rcgu.o" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.1nhu2gmvl07wlxhd.rcgu.o" "/LIBPATH:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\deps" "/LIBPATH:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-452d1b1b45f75fb9.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-5ceab4181bd1dfb1.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-ee63620c9a615dfc.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-1dced171ac40f317.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-e344794c4402c589.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-4a873b5445642af4.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-b457ffb0220370f3.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-3dcf84390a40067d.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-41cfe43f2c15c320.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-32a64f0768a3b3dc.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-0a994057c9b22b29.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-7ee144391913f95b.rlib" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-a1bb578584eeed3b.rlib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "advapi32.lib" "userenv.lib" "kernel32.lib" "libcmt.lib" "/NXCOMPAT" "/LIBPATH:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools\\release\\build\\ryu-c6f6c0e233049260\\build_script_build-c6f6c0e233049260.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\lib\\rustlib\\etc\\libstd.natvis"
  = note: LINK : fatal error LNK1318: Unexpected PDB error; OK (0) 'D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0-tools\release\build\ryu-c6f6c0e233049260\build_script_build-c6f6c0e233049260.pdb'
```

These link errors tend to be very difficult to diagnose (like #33145). I do not know if that is a transient error, as I haven't tried again.

Issue: gnu linker errors in run-make-fulldeps

75 of the run-make-fulldeps tests fail with the `gnu` target.  Lots of errors, probably all related to the same underlying problem of handling gnu vs windows paths.

```
D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsa-b-a-linker-guarda-b-a-linker-guard/b: No such file or directory

could not find native static library `foo`, perhaps an -L flag is missing?

D:\a\rust\rust\build\x86_64-pc-windows-gnu\test\run-make-fulldeps\arguments-non-c-like-enum\arguments-non-c-like-enum/libnonclike.a: No such file or directory

error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32-gcc" "-Wl,C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcgZbYmn\\list.def" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "C:\\MORE_SPACE\\x86_64-pc-windows-gnu\\stage2\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsc-dynamic-dylibc-dynamic-dylib\\foo.foo.172d8985-cgu.0.rcgu.o" "D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsc-dynamic-dylibc-dynamic-dylib\\foo.21ett6z3hgvj0xmp.rcgu.rmeta" "-L" "D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsc-dynamic-dylibc-dynamic-dylib" "-L" "C:\\MORE_SPACE\\x86_64-pc-windows-gnu\\stage2\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-lcfoo" "-Wl,--start-group" "-L" "C:\\MORE_SPACE\\x86_64-pc-windows-gnu\\stage2\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-lstd-6a8d029579976682" "-Wl,--end-group" "-Wl,-Bstatic" "C:\\MORE_SPACE\\x86_64-pc-windows-gnu\\stage2\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-94cb478f7048ff38.rlib" "-Wl,-Bdynamic" "-lkernel32" "-lws2_32" "-lbcrypt" "-ladvapi32" "-luserenv" "-lkernel32" "-lgcc_s" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "C:\\MORE_SPACE\\x86_64-pc-windows-gnu\\stage2\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-o" "D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsc-dynamic-dylibc-dynamic-dylib\\foo.dll" "-shared" "-Wl,--out-implib=D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsc-dynamic-dylibc-dynamic-dylib\\libfoo.dll.a" "-nodefaultlibs" "C:\\MORE_SPACE\\x86_64-pc-windows-gnu\\stage2\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: D:/a/rust/rust/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/6.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lcfoo
          collect2.exe: error: ld returned 1 exit status

cp: cannot stat 'D:\a\rust\rust\build\x86_64-pc-windows-gnu\test\run-make-fulldeps\extern-multiple-copies\extern-multiple-copies/libfoo1.rlib': No such file or directory

error: error writing dependencies to `D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepsinclude_bytes_depsinclude_bytes_deps\main.d`: The system cannot find the path specified. (os error 3)

/usr/bin/sh: line 1: D:arustrustbuildx86_64-pc-windows-gnutestrun-make-fulldepscompile-stdincompile-stdin/rust_out: No such file or directory
```

--- a/src/ci/github-actions/jobs.yml
+++ b/src/ci/github-actions/jobs.yml
@@ -25,10 +25,18 @@ runners:
     <<: *base-job
 
   - &job-windows-8c
-    os: windows-2019-8core-32gb
+    os: windows-2022-8core-32gb
     <<: *base-job
 
   - &job-windows-16c
+    os: windows-2022-16core-64gb
+    <<: *base-job
+
+  - &job-windows-2019-8c
+    os: windows-2019-8core-32gb
+    <<: *base-job
+
+  - &job-windows-2019-16c
     os: windows-2019-16core-64gb
     <<: *base-job
 
@@ -338,13 +346,13 @@ auto:
     env:
       RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-profiler
       SCRIPT: make ci-msvc
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c
 
   - image: i686-msvc
     env:
       RUST_CONFIGURE_ARGS: --build=i686-pc-windows-msvc
       SCRIPT: make ci-msvc
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c
 
   - image: x86_64-msvc-ext
     env:
@@ -352,7 +360,7 @@ auto:
       HOST_TARGET: x86_64-pc-windows-msvc
       RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-lld --save-toolstates=/tmp/toolstate/toolstates.json
       DEPLOY_TOOLSTATES_JSON: toolstates-windows.json
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c
 
   # 32/64-bit MinGW builds.
   #
@@ -403,7 +411,7 @@ auto:
         --set rust.codegen-units=1
       SCRIPT: python x.py build --set rust.debug=true opt-dist && PGO_HOST=x86_64-pc-windows-msvc ./build/x86_64-pc-windows-msvc/stage0-tools-bin/opt-dist windows-ci -- python x.py dist bootstrap --include-default-paths
       DIST_REQUIRE_ALL_TOOLS: 1
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c
 
   - image: dist-i686-msvc
     env:
@@ -415,7 +423,7 @@ auto:
         --enable-profiler
       SCRIPT: python x.py dist bootstrap --include-default-paths
       DIST_REQUIRE_ALL_TOOLS: 1
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c
 
   - image: dist-aarch64-msvc
     env:
@@ -426,7 +434,7 @@ auto:
         --enable-profiler
       SCRIPT: python x.py dist bootstrap --include-default-paths
       DIST_REQUIRE_ALL_TOOLS: 1
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c
 
   - image: dist-i686-mingw
     env:
@@ -460,4 +468,4 @@ auto:
     env:
       RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-extended --enable-profiler
       SCRIPT: python x.py dist bootstrap --include-default-paths
-    <<: *job-windows-8c
+    <<: *job-windows-2019-8c

